Add README-docker-compose.md and reference it from README.md and dock…

cwisecarver opened the pull request:
…er-compose.yml

<!-- Before submit your Pull Request, make sure you picked
     the right branch:

     - For hotfixes, select "master" as the target branch
     - For new features, select "develop" as the target branch
     - For release feature fixes, select the relevant release branch (release/X.Y.Z) as the target branch -->

## Purpose

<!-- Describe the purpose of your changes -->

## Changes

<!-- Briefly describe or list your changes  -->

## Side effects

<!--Any possible side effects? -->


## Ticket

<!-- Link to JIRA ticket, if applicable e.g. https://openscience.atlassian.net/browse/OSF-1234 -->

cwisecarver commented:
@icereval CR for me? I think it's all right. Links work, seems like English.

icereval commented on README-docker-compose.md:
```diff
@@ -0,0 +1,112 @@
+  - OSX
+    - Alias the loopback interface
+
+      `$ sudo ifconfig lo0 alias 192.168.168.167`
```
can you add a big note about running this on reboot

icereval commented:
Does this replace README.md? if not you'll want to update the `Running the OSF` section.

cwisecarver commented:
I added it [here](https://github.com/cwisecarver/osf.io/blob/feature/update-docker-compose-readme/README.md#table-of-contents), above `Running the OSF` and changed `Running the OSF` to `Running the OSF locally`.

cwisecarver commented:
I made it more explicit. https://github.com/cwisecarver/osf.io/blob/feature/update-docker-compose-readme/README.md#running-the-osf

cwisecarver commented:
Still need to add something about migrations, then I'm gtg.
